lib/rubocop/cop/rspec/sort_tags.rb

```diff
@@ -0,0 +1,333 @@
+          return unless (yellow_tape = investigator.yellow_tape)
+
+          add_offense(yellow_tape, message: investigator.report) do |corrector|
+            cleaner = CrimeSceneCleaner.new(tags, style)
+            corrector.replace(range(tags), cleaner.cleanup)
+          end
+        end
+
+        def range(tags)
+          range_between(
+            tags.first.loc.expression.begin_pos,
+            tags.last.loc.expression.end_pos
+          )
+        end
+
+        class Deputy # :nodoc:
+          def initialize(tags, style)
+            @sym_nodes = tags.last.hash_type? ? tags[0..-2] : tags
+            @hash_node = tags.last.hash_type? ? tags.last : nil
+            @style = style
+          end
+
+          private
+
+          def hash_pairs
+            @hash_node&.pairs || []
+          end
+        end
+
+        class CrimeSceneInvestigator < Deputy # :nodoc:
+          include RangeHelp
+
+          def initialize(tags, style, cop)
+            super(tags, style)
+            @cop = cop
+          end
+
+          def yellow_tape
+            first = first_node
+            last = last_node
+
+            return unless first && last
+
+            range_between(
+              first.loc.expression.begin_pos,
+              last.loc.expression.end_pos
+            )
+          end
+
+          def report
+            message = []
+
+            message << 'sort tags alphabetically' unless sorted?
+
+            if @style == :prefer_symbols
+              message << 'use `:tag` instead of `tag: true`'
+            end
+
+            if @style == :hash && @sym_nodes.any?
+              message << 'use `tag: true` instead of `:tag`'
+            end
+
+            message.join(' and ').concat('.').capitalize
+          end
+
+          private
+
+          def first_node
+            if include_symbols?
+              @sym_nodes.first
+            elsif include_hash?
+              return @hash_node unless hash_sorted? && @style == :prefer_symbols
+
+              hash_pairs.find do |pair_node|
+                pair_node.children.last.true_type?
+              end
+            end
+          end
+
+          def last_node
+            if include_hash?
+              return @hash_node unless hash_sorted? && @style == :prefer_symbols
+
+              hash_pairs.reverse.find do |pair_node|
+                pair_node.children.last.true_type?
+              end
+            elsif include_symbols?
+              @sym_nodes.last
+            end
+          end
+
+          def include_symbols?
+            if @style == :hash
+              @sym_nodes.any?
+            else
+              !symbols_sorted?
+            end
+          end
+
+          def include_hash?
+            return false unless @hash_node
+            return !hash_sorted? unless @style == :prefer_symbols
+
+            hash_pairs.any? do |pair_node|
+              pair_node.children.last.true_type?
+            end
```

Comment:
```suggestion
            hash_pairs.any? do |pair_node|
              pair_node.value.true_type?
            end
```
Or even
```suggestion
            hash_pairs.map(&:value).any?(&:true_type?)
```
or if you preserve the origal hash note, and not the pairs
```suggestion
            hash_node.each_value.any?(&:true_type?)
```
This applies to other parts of the code. Use the provided `key` and `value` methods instead of refering first and last children